### PR TITLE
New: Show rejection message if media info detects the file is an executable

### LINKED ISSUE #3331: Adjust DetectSample error to warn about malicious executable video files
**Describe the bug**
When Sonarr (latest v2 release) processes HEVC/h265 files, it doesn't automatically move the files. Instead, the files need "manual import".

If `mediainfo` is run against the file, I get the following:

```
/ # mediainfo --full /downloads/sonarr/downloaded-episode.mkv
General
Count                                    : 332
Count of stream of this kind             : 1
Kind of stream                           : General
Kind of stream                           : General
Stream identifier                        : 0
Complete name                            : /downloads/sonarr/downloaded-episode.mkv
Folder name                              : /downloads/sonarr
File name extension                      : downloaded-episode.mkv
File name                                : downloaded-episode
File extension                           : mkv
Format                                   : MZ
Format                                   : MZ
Format/Extensions usually used           : exe dll
Commercial name                          : MZ
Format profile                           : Executable / AMD x86-64
File size                                : 204765846
File size                                : 195 MiB
File size                                : 195 MiB
File size                                : 195 MiB
File size                                : 195 MiB
File size                                : 195.3 MiB
Stream size                              : 204765846
Stream size                              : 195 MiB (100%)
Stream size                              : 195 MiB
Stream size                              : 195 MiB
Stream size                              : 195 MiB
Stream size                              : 195.3 MiB
Stream size                              : 195 MiB (100%)
Proportion of this stream                : 1.00000
Encoded date                             : UTC 2019-04-27 20:03:30
File last modification date              : UTC 2019-08-08 00:05:23
File last modification date (local)      : 2019-08-08 00:05:23
FileExtension_Invalid                    : exe dll
```

**Logs**
Link to debug or trace log files.

```
2019-10-09 21:02:08.294907500  [sonarr] [Error] DetectSample: [/downloads/sonarr/downloaded-episode.mkv] has a runtime of 0, is it a valid video file? 
```

**System Information**
 - Problem exists in the `linuxserver/sonarr` container

I thought that the problem may have been with the version of MediaInfoLib in the `linuxserver/sonarr` container.

I built Sonarr manually with the latest version of MediaInfoLib (v19.09), however the problem still seems to exist.

Thanks.

### PR DESCRIPTION
**Pull request by markus101**
#### Database Migration
NO

#### Description
Uses the format from media info (newly fetched) to determine if the file is an executable and rejects imports with an appropriate message. Also cleans up fetching processing of media info so it's not done up to 3 times, files left over after importing may get processed a second time, but previously it was done twice for every file processed during import.

#### Issues Fixed or Closed by this PR

* Closes #3331

### CODE REVIEW

File: src/NzbDrone.Core/MediaFiles/EpisodeImport/DetectSampleResult.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@ public enum DetectSampleResult
     {
         Indeterminate,
         Sample,
-        NotSample
+        NotSample,
+        Executable
```
Comment (Taloth) [inline src/NzbDrone.Core/MediaFiles/EpisodeImport/DetectSampleResult.cs:8]: I'm ok with it, but technically speaking this isn't sample detection, it's IDetectMalware or something.

File: src/NzbDrone.Core/MediaFiles/EpisodeImport/DetectSample.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@ namespace NzbDrone.Core.MediaFiles.EpisodeImport
 {
     public interface IDetectSample
     {
-        DetectSampleResult IsSample(Series series, string path, bool isSpecial);
+        DetectSampleResult IsSample(Series series, string path, bool isSpecial, MediaInfoModel mediaInfo = null);
```
Comment (Taloth) [inline src/NzbDrone.Core/MediaFiles/EpisodeImport/DetectSample.cs:11]: We need a mediainfo cache by path+size+lastmod. Not now, but it'd eliminate the need for the optional param.

**Review by Taloth — COMMENTED** [review]
Should've reviewed this months ago. Personally I think this should be a separate IDetectMalware or something like it, but I'm ok with it as-is since refactoring it would take unnecessary time.

**Comment by markus101** [comment]
Closing, moving to ffmpeg and will re-evaluate this.

**Pull request closed without merging**